Initial release for Leybold IM540 gauge controller

janezg-SLAC opened the pull request:
This EPICS IOC controls the Leybold IM540 Gauge controller. 
IOC integrates all needed commands and readbacks. This was tested on controller in TMO hutch for DREAM end station main chamber gauge.

ZLLentz commented:
> This was tested on controller in TMO hutch for DREAM end station main chamber gauge.

I'm going to trust your testing and mostly review for structural/procedural things rather than for functionality

ZLLentz commented on .gitignore:
Suggestion: consider adding the following items I was putting on a standard gitignore:
```
# Terminal editor ignores
*.swp
*~

# IDE ignores
*.vscode

# Core dumps
core.*
```

Mostly editor stuff + you have no idea how often people accidentally commit core dumps, it's crazy

janezg-SLAC replied:
Will FIX.

ZLLentz commented on README:
Note that this repo also has a README.md

Suggestion: remove this and add a link to README.md to a pertinent docs page about the device itself.

janezg-SLAC replied:
Confluence page right? https://confluence.slac.stanford.edu/spaces/PCDS/pages/341250673/Leybold+IM+540

ZLLentz replied:
confluence page is a great choice :+1:

ZLLentz commented on app/Db/leyboldim540_ch14.template:
```diff
@@ -0,0 +1,397 @@
+#   CHANNEL NUMBER
+#
+#
+#record(mbbi, "$(DEV):GET_OFC_CH$(CH)")
```
Question: I notice many of these records are commented out. Is this scope for later or can they be removed?

janezg-SLAC replied:
Yes, that's true there are many commands commented out. Some got commented out because are related only to controller's CH3 and CH4 which, based on Jing's comment will not be used (CH3 and CH4 supports different sensor then CH1 and CH2). Some got commented out because were not needed and because I was getting random timeouts on PVs and that's why I scaled down the number of commands a bit.

ZLLentz commented on children/ioc-im540-test.cfg:
```diff
@@ -0,0 +1,8 @@
+ARCH=rhel7-x86_64
+
+ENGINEER="enginner_name (shortname)"
+LOCATION="Somewhere Over the Rainbow"
```
General comment: :+1:, great location placeholder

ZLLentz commented on iocBoot/templates/edm-ioc.cmd:
Nitpick: the template expander also now supports `pydm-` and `launchgui-` in addition to `edm-`. Maybe this can use one of those since it isn't an `edm` screen.

janezg-SLAC replied:
Will Fix. Makes sense now. When I started this IOC didn't have this option :D

ZLLentz reviewed (commented):
I reviewed everything except the ui files, which I'll look at after sending this.

Everything in the code/configuration looks clean and well-organized. I appreciate the various in-line comments etc. where you've added those.

I made a few small suggestions and asked a question. You should respond to those and either edit or decide not to edit, and then I can approve.

Side-note: I'd appreciate it if you tried to compile this on psbuild-rocky9 too, and write down if it succeeds or fails. This would help the ongoing rocky9 migration that @janeliu-slac is working on. (https://confluence.slac.stanford.edu/spaces/PCDS/pages/573790398/Rocky+9+Common+IOCs+Migration+Status)

ZLLentz commented:
Overall, good job. I'll approve after changes or decisions not to make changes, either way is ok.

janezg-SLAC commented on im540Screens/im540.ui:
/reg/g/pcds/epics-dev/janezg/tmo/im540/current/build/iocBoot/ioc-im540-tst/edm-ioc-im540-tst.cmd

Here you can see working IOC GUI.

ZLLentz replied:
Looks even better with the connections!

janezg-SLAC commented:
Ok, I think I updated everything as requested. I also updated the config and release_site files and was able to build for rocky9. Went back and built for rhel7 which was also successful. After that i built again children IOC, rebooted the test IOC and test the IOC with device. Everything seems to work.

ZLLentz commented:
I'll pull again, verify the `make`s, then approve. Thank you for your efforts!

ZLLentz reviewed (approved):
Looks great! Let's get this in!
